### PR TITLE
KOGITO-5789 - consider Maven reactor in kogito-quarkus-test-list

### PR DESCRIPTION
**Pull request by jstastny-cz**
https://issues.redhat.com/browse/KOGITO-5789

This PR attempts at fixing Maven reactor related issues with kogito-quarkus-test-list.

Goal is to
* enable dynamic "subscription" of integration-test modules, so that only those which are built as part of the reactor are included into the resulting XML test list.
* Adhere to `excludeITModules` profile, not to build (and then release) kogito-quarkus-test-list when the integration-test modules were excluded.

Implemented as
* Each applicable integration-test module need to activate rpkgtests-maven-plugin execution for itself by adding
  ```
             <plugin> <!-- this way module defines itself as included in kogito-quarkus-test-list artifact -->
                <groupId>org.l2x6.rpkgtests</groupId>
                <artifactId>rpkgtests-maven-plugin</artifactId>
            </plugin>
  ```
* This writes a simple test list into temporary location
* In kogito-quarkus-test-list module, merge these temporary XML files, using XSLT transformation and produce the final artifact.
* Attach the resulting XML file to the released artifact.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Run all builds</b>  
  Please add comment: <b>Jenkins retest this</b>

* <b>Run (or rerun) specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] tests</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Run (or rerun) LTS specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

* <b>Run (or rerun) native specific test(s)</b>  
  Please add comment: <b>Jenkins (re)run [runtimes|optaplanner|apps|examples] native</b>

* <b>Full Kogito testing</b> (with cloud images and operator BDD testing)  
  Please add comment: <b>Jenkins run BDD</b>  
  <b>This check should be used only if a big change is done as it takes time to run, need resources and one full BDD tests check can be done at a time ...</b>
</details>

### CODE REVIEW

**Comment by jstastny-cz** [comment]
Waiting for build outcomes before turning to regular PR. Reactor order will be decisive point (the kogito-quarkus-test-list needs to run after all integration-test modules - ideally as the last one).

File: kogito-quarkus-test-list/pom.xml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+          <!-- needs to be in sync with rpkg.plugin.working.dir property -->
+          <exists>${java.io.tmpdir}/test-list-fragments/</exists>
+        </file>
+      </activation>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.codehaus.mojo</groupId>
+            <artifactId>xml-maven-plugin</artifactId>
+            <version>1.0.2</version>
```
Comment (jstastny-cz) [inline kogito-quarkus-test-list/pom.xml:40]: need to use the already defined property from kogito-build-parent.

File: kogito-quarkus-test-list/pom.xml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                    </paremeter>
+                  </parameters>
+                </transformationSet>
+              </transformationSets>
+            </configuration>
+            <dependencies>
+              <dependency>
+                <groupId>net.sf.saxon</groupId>
+                <artifactId>saxon</artifactId>
+                <version>8.7</version>
```
Comment (jstastny-cz) [inline kogito-quarkus-test-list/pom.xml:67]: Need to parameterize, not sure where to put. Kogito-build-parent seems to have just maven-plugin dependencies now, kogito-dependencies-bom defining the others. But this is solely for use by the plugin, so putting into kogito-dependencies-bom could pollute the BOM. @cristianonicolai wdyt?

File: integration-tests/integration-tests-quarkus-norest/pom.xml [hunk 3]
```diff
@@ -128,6 +128,10 @@
           </systemPropertyVariables>
         </configuration>
       </plugin>
+      <plugin> <!-- this way module defines itself as included in kogito-quarkus-test-list artifact -->
```
Comment (jstastny-cz) [inline integration-tests/integration-tests-quarkus-norest/pom.xml:131]: This module was actually excluded before, need to remove this activation for now.

File: integration-tests/integration-tests-quarkus-processes/pom.xml [hunk 4]
```diff
@@ -184,6 +184,10 @@
           </systemPropertyVariables>
         </configuration>
       </plugin>
+      <plugin> <!-- this way module defines itself as included in kogito-quarkus-test-list artifact -->
```
Comment (jstastny-cz) [inline integration-tests/integration-tests-quarkus-processes/pom.xml:187]: Same here, processes were excluded before.

File: quarkus/extensions/kogito-quarkus-decisions-extension/kogito-quarkus-decisions-integration-test-hot-reload/pom.xml [hunk 5]
```diff
@@ -75,6 +75,10 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin> <!-- this way module defines itself as included in kogito-quarkus-test-list artifact -->
```
Comment (jstastny-cz) [inline quarkus/extensions/kogito-quarkus-decisions-extension/kogito-quarkus-decisions-integration-test-hot-reload/pom.xml:78]: This also needs to be moved out.

File: quarkus/extensions/kogito-quarkus-processes-extension/kogito-quarkus-processes-integration-test-hot-reload/pom.xml [hunk 6]
```diff
@@ -81,6 +81,10 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin> <!-- this way module defines itself as included in kogito-quarkus-test-list artifact -->
```
Comment (jstastny-cz) [inline quarkus/extensions/kogito-quarkus-processes-extension/kogito-quarkus-processes-integration-test-hot-reload/pom.xml:84]: to be excluded.

File: quarkus/extensions/kogito-quarkus-rules-extension/kogito-quarkus-rules-integration-test-hot-reload/pom.xml [hunk 7]
```diff
@@ -80,6 +80,10 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin> <!-- this way module defines itself as included in kogito-quarkus-test-list artifact -->
```
Comment (jstastny-cz) [inline quarkus/extensions/kogito-quarkus-rules-extension/kogito-quarkus-rules-integration-test-hot-reload/pom.xml:83]: to be excluded.

**Review by jstastny-cz — COMMENTED** [review]
Changes in scope of included modules are needed to be in sync with https://issues.redhat.com/browse/KOGITO-5344.

File: kogito-quarkus-test-list/pom.xml (hunk 2 above)
Comment (cristianonicolai) [reply kogito-quarkus-test-list/pom.xml]: build-parent seems fine to me. @ricardozanini wdyt?

File: integration-tests/integration-tests-quarkus-processes-persistence/integration-tests-processes-persistence-common/pom.xml [hunk 8]
```diff
@@ -65,6 +65,10 @@
           </execution>
         </executions>
       </plugin>
+      <plugin> <!-- this way module defines itself as included in kogito-quarkus-test-list artifact -->
```
Comment (cristianonicolai) [inline integration-tests/integration-tests-quarkus-processes-persistence/integration-tests-processes-persistence-common/pom.xml:68]: @jstastny-cz this one doesnt have any runnable test in it, just common resources used by other modules, not sure it needs to be included.

File: kogito-quarkus-test-list/pom.xml [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+  <artifactId>kogito-quarkus-test-list</artifactId>
+  <name>Kogito :: Quarkus Test List</name>
+  <description>Publishes a list of integration tests in XML format that should be run by the Quarkus Platform
+  </description>
+
+  <packaging>pom</packaging>
+
+  <properties>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
```
Comment (cristianonicolai) [inline kogito-quarkus-test-list/pom.xml:22]: this is already set in the parent

File: kogito-quarkus-test-list/pom.xml [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <parent>
+    <groupId>org.kie.kogito</groupId>
+    <artifactId>kogito-build-parent</artifactId>
+    <version>2.0.0-SNAPSHOT</version>
+    <relativePath>../kogito-build/kogito-build-parent/pom.xml</relativePath>
+  </parent>
+  <modelVersion>4.0.0</modelVersion>
+
+  <artifactId>kogito-quarkus-test-list</artifactId>
```
Comment (cristianonicolai) [inline kogito-quarkus-test-list/pom.xml:13]: @jstastny-cz if you add an artifical dependency to the integration-tests POM, would that be enough to keep this module under the Quarkus artefacts? That way the build reactor would make sure is only build afterwards.

File: integration-tests/integration-tests-quarkus-processes-persistence/integration-tests-processes-persistence-common/pom.xml (hunk 8 above)
Comment (jstastny-cz) [reply integration-tests/integration-tests-quarkus-processes-persistence/integration-tests-processes-persistence-common/pom.xml:68]: Thanks. I will remove this one.

File: kogito-quarkus-test-list/pom.xml (hunk 10 above)
Comment (jstastny-cz) [reply kogito-quarkus-test-list/pom.xml:13]: I tried that with dependency on `integration-tests` of type `pom`, the reactor then looked like
```
integration-tests/pom.xml
quarkus/extensions/kogito-quarkus-test-list/pom.xml
...
integration-tests/*/pom.xml
```

File: kogito-quarkus-test-list/pom.xml (hunk 9 above)
Comment (jstastny-cz) [reply kogito-quarkus-test-list/pom.xml:22]: Removing.

**Review by cristianonicolai — APPROVED** [review]

**Comment by jstastny-cz** [comment]
Does not work properly, kogito-quarkus-test-list is getting into test-list itself. I will add tests for the test-list as part of this (should be quick).

**Comment by jstastny-cz** [comment]
Moved back to draft. The changes in rpkgtests-maven-plugin https://github.com/ppalaga/rpkgtests-maven-plugin/pull/14 was merged and version 0.11.0 released. This will simplify configuration, I will adapt this PR once it's available in maven central.

**Comment by jstastny-cz** [comment]
Closing this one and will create a new PR, to limit scope of reviews (new config should be much simpler).

**Comment by jstastny-cz** [comment]
Superseded by #1584

**Pull request closed without merging**